### PR TITLE
Golden file testing

### PR DESCRIPTION
**Pull request by EwenQuim**
Adds golden file testing, and vacuum for schema validation.

This helps tracks the generated OpenAPI schema when making change on the generation

### CODE REVIEW

File: mux.go [hunk 1]
```diff
@@ -125,7 +125,7 @@ func Register[T, B any](s *Server, route Route[T, B], controller http.Handler, m
 
 	route.Operation.Summary = route.NameFromNamespace(camelToHuman)
 	route.Operation.Description = "controller: `" + route.FullName + "`\n\n---\n\n"
-	route.Operation.OperationID = route.Method + " " + s.basePath + route.Path + ":" + route.NameFromNamespace()
+	route.Operation.OperationID = route.Method + "_" + s.basePath + strings.ReplaceAll(strings.ReplaceAll(route.Path, "{", ":"), "}", "")
```
Comment (EwenQuim) [inline mux.go:128]: This is a breaking change fix, will notify it on next release!

File: openapi.go [hunk 2]
```diff
@@ -162,8 +163,8 @@ func RegisterOpenAPIOperation[T, B any](s *Server, method, path string) (*openap
 
 	// Request body
 	bodyTag := schemaTagFromType(s, *new(B))
-	if (method == http.MethodPost || method == http.MethodPut || method == http.MethodPatch) && bodyTag.name != "unknown-interface" && bodyTag.name != "string" {
-		content := openapi3.NewContentWithSchemaRef(&bodyTag.SchemaRef, []string{"application/json"})
+	if bodyTag.name != "unknown-interface" {
+		content := openapi3.NewContentWithSchemaRef(&bodyTag.SchemaRef, []string{"application/json", "application/xml"})
```
Comment (EwenQuim) [inline openapi.go:167]: Thanks to #137 we will have this behaviour accepted as well 🥳

File: openapi.go [hunk 3]
```diff
@@ -192,6 +193,10 @@ func RegisterOpenAPIOperation[T, B any](s *Server, method, path string) (*openap
 	for _, pathParam := range parsePathParams(path) {
 		parameter := openapi3.NewPathParameter(pathParam)
 		parameter.Schema = openapi3.NewStringSchema().NewRef()
+		if strings.HasSuffix(pathParam, "...") {
```
Comment (EwenQuim) [inline openapi.go:196]: Supports `{things...}` syntax https://pkg.go.dev/net/http#ServeMux

File: mux.go (hunk 1 above)
Comment (dylanhitt) [reply mux.go:131]: Much better 👍

**Review by dylanhitt — APPROVED** [review]

**Comment by dylanhitt** [comment]
My only comment is we should probably add a check in the CI to ensure the golden-file is being updated. The CI check shouldn't update the file, but it would just fail until the author of the MR updated it.

**Comment by EwenQuim** [comment]
If the golden file isn't updated and it should, the tests will fail, so no issue here :)

**Pull request merged**